File: game/scripts/vscripts/modifiers/upgrades/items/modifier_upgrade_item_boots_of_elves.lua
modifier_upgrade_item_boots_of_elves = class({})

function modifier_upgrade_item_boots_of_elves:RemoveOnDeath() return false end
function modifier_upgrade_item_boots_of_elves:IsPurgable() return false end

function modifier_upgrade_item_boots_of_elves:OnCreated()
    self.damage = 1
	self.as_speed = 25
end

function modifier_upgrade_item_boots_of_elves:DeclareFunctions()
    return { 
        MODIFIER_PROPERTY_PREATTACK_BONUS_DAMAGE,
        MODIFIER_PROPERTY_ATTACKSPEED_BONUS_CONSTANT,
    }
end

function modifier_upgrade_item_boots_of_elves:GetModifierAttackSpeedBonus_Constant()
	return self.as_speed
end

function modifier_upgrade_item_boots_of_elves:GetModifierPreAttack_BonusDamage(params)
    return self.damage
end

function modifier_upgrade_item_boots_of_elves:GetTexture(params)
    return 'item_boots_of_elves'
end

File: src/vscripts/modifiers/upgrades/items/modifier_upgrade_item_boots_of_elves.ts
import { registerModifier } from "../../../lib/dota_ts_adapter";
import { ModifierUpgrade } from "../modifier_upgrade";

@registerModifier("modifier_upgrade_item_boots_of_elves")
class ModifierItemBootsOfElves extends ModifierUpgrade {
    damage = 1;
    asSpeed = 25;

    DeclareFunctions() {
        return [ModifierFunction.PREATTACK_BONUS_DAMAGE, ModifierFunction.ATTACKSPEED_BONUS_CONSTANT];
    }

    GetModifierPreAttack_BonusDamage() {
        return this.damage;
    }

    GetModifierAttackSpeedBonus_Constant() {
        return this.asSpeed;
    }

    GetTexture() {
        return "item_boots_of_elves";
    }
}
